### PR TITLE
Add CLI support to create and update properties for Catalog and Schema

### LINKED ISSUE #175: Add server support for updating properties in catalog and schema
**Is your feature request related to a problem? Please describe.**

Currently, the update operations for catalog and schema do not support updating properties.

**Describe the solution you would like**

The `UpdateCatalog`/`UpdateSchema` objects can already take in properties. We would like to take these properties and overwrite any existing properties the entity has.
For example:
if we have a catalog: `catalog1` with properties `{prop1 -> value1, prop2 -> value2}`
and we call the update endpoint with new properties `{prop2 -> value4, prop3 -> value3}`
`catalog1` should have the properties `{prop2 -> value4, prop3 -> value3}`

**Additional context**

Rough idea on how to do this for catalog (schema would be similar)
- If properties is supplied in the UpdateCatalog object, [`updateCatalog`](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java#L117) must remove existing properties and update those with the new properties supplied.
- This [function](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/main/java/io/unitycatalog/server/persist/PropertyRepository.java#L16) is used to interact with properties.
- Corresponding test cases must be added [here](https://github.com/unitycatalog/unitycatalog/blob/main/server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java#L59) to ensure behaviour is as expected.

Please feel free to reach out on [slack](https://go.unitycatalog.io/slack) if you need more details

### LINKED ISSUE #176: Add CLI support to provide properties for catalog and schema
**Is your feature request related to a problem? Please describe.**

Currently, there is no way to provide properties for catalog and schema via the CLI

**Describe the solution you would like**

We should be able to provide a json format of a map of properties in the CLI for create and update operations. Ex: `--properties '{"key1": "value1", "key2": "value2"}'`

**Additional context**

- This is already implemented for tables [here](https://github.com/unitycatalog/unitycatalog/blob/ed2ad4af6fbd663e45a0271c2a750d9e2acab9eb/examples/cli/src/main/java/io/unitycatalog/cli/TableCli.java#L86). We should try to expand on the same and reuse the same code.
- We would also need to correspondingly update the optional and required params [here](https://github.com/unitycatalog/unitycatalog/blob/ed2ad4af6fbd663e45a0271c2a750d9e2acab9eb/examples/cli/src/main/java/io/unitycatalog/cli/utils/CliUtils.java#L76).
- Preferably add test cases that ensure feature completeness. For example [here](https://github.com/unitycatalog/unitycatalog/blob/ed2ad4af6fbd663e45a0271c2a750d9e2acab9eb/examples/cli/src/test/java/io/unitycatalog/cli/catalog/CliCatalogOperations.java#L28).
- This should also unblock us to fix [these](https://github.com/unitycatalog/unitycatalog/blob/ed2ad4af6fbd663e45a0271c2a750d9e2acab9eb/server/src/test/java/io/unitycatalog/server/base/catalog/BaseCatalogCRUDTest.java#L24) TODOs.

Please feel free to reach out on [slack](https://go.unitycatalog.io/slack) if you need additional details.

### PR DESCRIPTION
**Pull request by vksx**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

- Added CLI support to create properties for Catalog and Schema:
  - Cataog: 
    ```sh
    bin/uc catalog create --name catalog_name --properties '{"key1": "value1", "key2": "value2"}'
    ```
  - Schema: 
    ```sh
    bin/uc schema create --catalog catalog_name --name schema_name  --properties '{"key1": "value1", "key2": "value2"}'
     ```
- Added CLI support to update properties for Catalog and Schema:
  - Catalog:
    ```sh
    bin/uc catalog update --name catalog_name --properties '{"new_key1": "new_value1", "new_key2": "new_value2"}'
    ```
  - Schema:
    ```sh
    bin/uc schema update --full_name catalog_name.schema_name --properties '{"new_key1": "new_value1", "new_key2": "new_value2"}'
    ```

- Documentation changes.
  - Added properties usage for Catalog and Schema
  - Added headings in the documentation to make it more presentable
  - Fixed typos and inconsistencies

Closes #175 and #176

### CODE REVIEW

**Comment by vksx** [comment]
@vikrantpuppala The changes are done - please review them once you get a chance. Thx! 🙏

**Comment by vksx** [comment]
@MrPowers Should we add `uc_docs_venv` to `.gitignore` as well?

**Comment by xx-db** [comment]
CLI changes LGTM. Nice work. Server changes appear to overlap with #190 which already has some comments in it. Can you please review #190 and adjust accordingly?

**Comment by vksx** [comment]
Thanks for reviewing this! 

Actually, I was under the impression that we need to add support for creating and updating Catalog and Schema in #176 - and that required server changes as well hence this covers all of it along with the relevant test cases.

And, I was unaware of #175 and that it was being worked on - my bad.

Apparently, this PR now fixes both #175 and #176 - so please let me know the next steps.

**Comment by xx-db** [comment]
Thank you for the update! I want to clarify that since #190 has been open for a long time and will be merged soon, I would suggest removing the duplicated parts to keep the scope of this PR to #176.

**Comment by vksx** [comment]
@xx-db Server file changes are reverted to keep the scope to #176. PTAL.

**Review by xx-db — APPROVED** [review]
Thanks for updating! LGTM

**Comment by alexreid-db** [comment]
Do we need to wait for https://github.com/unitycatalog/unitycatalog/pull/190 to merged in before merging this in?

**Review by vikrantpuppala — APPROVED** [review]

**Comment by vksx** [comment]
@vikrantpuppala added @tlm365's commits to this PR. PTAL.

File: server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java [hunk 1]
```diff
@@ -55,6 +55,8 @@ public CatalogInfo addCatalog(CreateCatalog createCatalog) {
         CatalogInfoDAO catalogInfoDAO = CatalogInfoDAO.from(catalogInfo);
         PropertyDAO.from(catalogInfo.getProperties(), catalogInfoDAO.getId(), Constants.CATALOG)
             .forEach(session::persist);
+        RepositoryUtils.attachProperties(
```
Comment (vikrantpuppala) [inline server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java]: this isn't needed since it is set at line 46

**Review by vikrantpuppala — COMMENTED** [review]
one minor comment

File: server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java (hunk 1 above)
Comment (vikrantpuppala) [reply server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java]: seems like you just addressed this, thanks!

**Review by vikrantpuppala — APPROVED** [review]
@vksx thanks for finishing up this long pending issue.
@ramesh-db could you merge this please.

**Comment by vikrantpuppala** [comment]
Closes #176 and #175

**Review by alexreid-db — APPROVED** [review]

**Pull request merged**